### DISCUSSION TITLE
DAB - Column type DateTimeOffset not yet supported

### DISCUSSION THREAD

**Discussion by nihasalangaden** [Azure/data-api-builder#1356] [General]
Hi All,

I was trying this with an existing table , but getting the below error: any thoughts?

dbug: Azure.DataApiBuilder.Service.Services.ISqlMetadataProvider[0]
      Logging primary key information for entity: test.
dbug: Azure.DataApiBuilder.Service.Services.ISqlMetadataProvider[0]
      Primary key column name: application_id
      Primary key mapped name: application_id
      Type: Byte
      IsNullable: False
      IsAutoGenerated: False
fail: Azure.DataApiBuilder.Service.Startup[0]
      Unable to complete runtime initialization operations due to:
System.ArgumentException: Column type DateTimeOffset not yet supported.
         at Azure.DataApiBuilder.Service.Parsers.EdmModelBuilder.BuildEntityTypes(ISqlMetadataProvider sqlMetadataProvider)
         at Azure.DataApiBuilder.Service.Parsers.EdmModelBuilder.BuildModel(ISqlMetadataProvider sqlMetadataProvider)
         at Azure.DataApiBuilder.Service.Parsers.ODataParser.BuildModel(ISqlMetadataProvider sqlMetadataProvider)
         at Azure.DataApiBuilder.Service.Services.SqlMetadataProvider`3.InitODataParser()
         at Azure.DataApiBuilder.Service.Services.SqlMetadataProvider`3.InitializeAsync()
         at Azure.DataApiBuilder.Service.Startup.PerformOnConfigChangeAsync(IApplicationBuilder app)
info: Azure.DataApiBuilder.Service.Startup[0]
      Loading config file: dab-config.json
fail: Azure.DataApiBuilder.Service.Startup[0]
      Exiting the runtime engine...
**Table structure is as below:**
application_id	tinyint
application_cd	varchar
application_nm	varchar
create_ts	datetimeoffset
update_ts	datetimeoffset

Config file added like below:
"test": {
        "source": {
          "type": "table",
          "object": "dbo.application_ref",
          "key-fields":["application_id"]
        },
        "rest": {
          "methods": [ "get" ]
        },
        "graphql": {
          "operation": "query"
        },
        "permissions": [{
         "role": "anonymous",
          "actions": [ "*" ]
        }]
      }

**Comment by Aniruddh25** [comment]
Thanks for reporting this issue @nihasalangaden!

It does look like this case was not supported yet - here as per the stack trace: 
https://github.com/Azure/data-api-builder/blob/2107b7d730a9d229d52aed957b8f92a48e459bc4/src/Service/Parsers/EdmModelBuilder.cs#L112

While a new release with the fix would be out soon, we encourage you to contribute as well following: https://github.com/Azure/data-api-builder/blob/main/CONTRIBUTING.md

**Reply by Aniruddh25** [reply]
Created issue: https://github.com/Azure/data-api-builder/issues/1358 to track this.
